src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -827,15 +857,6 @@ public void onSaveAllChunks(boolean saveAllChunks, IProgressUpdate progressCallb
         }
     }
 
-    @Inject(method = "addBlockEvent", at = @At("HEAD"), cancellable = true)
-    public void addBlockEvent(BlockPos pos, Block blockIn, int eventID, int eventParam, CallbackInfo ci) {
-        // We fire a Pre event to make sure our captures do not get stuck in a loop.
-        // This is very common with pistons as they add block events while blocks are being notified.
-        if (SpongeCommonEventFactory.handleChangeBlockEventPre(this, pos)) {
```

Comment:
*faceplant*, I moved this event over to onAddBlockEvent to prevent event spam. (which is after the check for if it's a duplicate event)